shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java

```diff
@@ -135,15 +135,10 @@ private void initialize() {
 
 	@Override
 	public ResponseEntity<Void> deleteDataset(
-			@Parameter(name = "id of the dataset", required = true) @PathVariable("datasetId") final Long datasetId)
-					throws RestServiceException {
+			@Parameter(name = "id of the dataset", required = true) @PathVariable("datasetId") final Long datasetId) {
 		try {
 			datasetService.deleteById(datasetId);
 			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-		} catch (EntityNotFoundException e) {
-			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-		} catch (RestServiceException e) {
-			return new ResponseEntity<>(HttpStatus.UNPROCESSABLE_ENTITY);
 		} catch (Exception e) {
 			LOG.error("Error while deleting dataset. Please check DICOM server configuration.", e);
 			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```

Comment:
Not your code obviously, but I think it should be more logic to return "HttpStatus.INTERNAL_SERVER_ERROR" as in deleteDatasets and not the same HTTP response as if the deletion was a success

Reply:
Hi @jcomedouteau,

Thx for your remarks.
I've harmonized exception management for both methods.

In fact, `EntityNotFoundException` | `RestServiceException` | `Exception` will be intercepted in class `GlobalExceptionHandler` of ms-common, and properly returned as a `ResponseEntity<>`.

So for both method :
- I rethrow `EntityNotFoundException` | `RestServiceException` 
- I encapsulate others exception in a `RestServiceException` and throw it.

All right for you ?